unblob/handlers/executable/elf.py

```diff
@@ -211,11 +211,11 @@ def get_end_offset(
             start_offset + header.e_phoff + (header.e_phnum * header.e_phentsize)
         )
 
-        last_section_end = self.get_last_section_end(
+        last_section_end = start_offset + self.get_last_section_end(
             file, start_offset + header.e_shoff, header.e_shnum, endian
         )
 
-        last_program_end = self.get_last_program_end(
+        last_program_end = start_offset + self.get_last_program_end(
```

Comment:
With this change, every component has a `start_offset` added, so what about rewriting the function like this?
```py
    def get_end_offset(
        self, file: File, start_offset: int, header: Instance, endian
    ) -> int:
        # Usually the section header is the last, but in some cases the program headers are
        # put to the end of the file, and in some cases sections header and actual sections
        # can be also intermixed, so we need also to check the end of the last section and
        # also the last program segment.
        # We check which one is the last and use it as a file size.
        section_headers_end = (
            header.e_shoff + (header.e_shnum * header.e_shentsize)
        )
        program_headers_end = (
            header.e_phoff + (header.e_phnum * header.e_phentsize)
        )

        last_section_end = self.get_last_section_end(
            file, start_offset + header.e_shoff, header.e_shnum, endian
        )

        last_program_end = self.get_last_program_end(
            file, start_offset + header.e_phoff, header.e_phnum, endian
        )

        return start_offset + max(
            section_headers_end, program_headers_end, last_section_end, last_program_end
        )
```

Reply:
done